platform/ui/src/components/SegmentationGroupTable/SegmentationGroupSegment.tsx

```diff
@@ -21,200 +21,159 @@ const SegmentItem = ({
   onToggleVisibility,
   onToggleLocked,
 }) => {
-  const [isHovering, setIsHovering] = useState(false);
-  const [isSegmentIndexHovering, setIsSegmentIndexHovering] = useState(false);
-
-  const onMouseEnter = () => setIsHovering(true);
-  const onMouseLeave = () => setIsHovering(false);
+  const [isRowHovering, setRowIsHovering] = useState(false);
+  const [isNumberBoxHovering, setIsNumberBoxHovering] = useState(false);
 
   const cssColor = `rgb(${color[0]},${color[1]},${color[2]})`;
 
   return (
     <div
-      className={classnames(
-        'bg-primary-dark group relative flex cursor-pointer overflow-hidden border text-[12px] transition duration-300',
-        {
-          'border-primary-light rounded-[3px]': isActive,
-        },
-        {
-          'border-transparent': !isActive,
-        }
-      )}
-      onMouseEnter={onMouseEnter}
-      onMouseLeave={onMouseLeave}
+      className={classnames('text-aqua-pale group/row flex min-h-[28px] bg-black')}
+      onMouseEnter={() => setRowIsHovering(true)}
+      onMouseLeave={() => setRowIsHovering(false)}
       onClick={e => {
         e.stopPropagation();
         onClick(segmentationId, segmentIndex);
       }}
-      role="button"
       tabIndex={0}
       data-cy={'segment-item'}
     >
       <div
-        className={classnames(
-          'flex w-[27px] items-center justify-center border-r border-r-black text-[12px]',
-          {
-            'bg-primary-light border-l-primary-light rounded-l-sm border-l text-black': isActive,
-            'rounded-l-sm  border-l border-l-[#1d3c58] bg-[#1d3c58]': !isActive && isHovering,
-            'bg-primary-dark text-aqua-pale': !isActive && !isHovering && isVisible,
-            'bg-[#140e2e] text-[#537594] opacity-60': !isActive && !isVisible,
-          }
-        )}
-        onMouseEnter={() => setIsSegmentIndexHovering(true)}
-        onMouseLeave={() => setIsSegmentIndexHovering(false)}
+        className={classnames('bg-primary-dark group/number grid w-[32px] place-items-center', {
+          'bg-primary-light border-primary-light rounded-l-[4px] border text-black': isActive,
+          'border-primary-dark border': !isActive,
+        })}
+        onMouseEnter={() => setIsNumberBoxHovering(true)}
+        onMouseLeave={() => setIsNumberBoxHovering(false)}
       >
-        {isSegmentIndexHovering && showSegmentDelete ? (
+        {isNumberBoxHovering && showDelete ? (
```

Comment:
Not a big deal, but I believe instead of using `onMouseEnter` and `onMouseLeave` to keep track of the hover we could also add a `group` class to the parent `div` and then have class names `hidden group-hover:block` on the `Icon`.